libs/net/core/Extensions/StringExtensions.cs

```diff
@@ -272,41 +286,41 @@ public static bool DirectoryExists(this string path, bool ignoreCase = false)
     /// <returns>formatted body string</returns>
     public static string ConvertTextToParagraphs(string? text, string paragraphRegex)
     {
-        string sanitizedString = string.IsNullOrEmpty(text) ? string.Empty : text;
+        string value = string.IsNullOrEmpty(text) ? string.Empty : text;
 
-        if (!string.IsNullOrEmpty(sanitizedString))
+        if (!string.IsNullOrEmpty(value))
         {
             // pattern for any matching pair of tags
-            Regex tagRegex = new Regex(@"<\s*([^ >]+)[^>]*>.*?<\s*/\s*\1\s*>");
+            Regex tagRegex = HasHtmlRegex();
             const string PARAGRAPH_MARKER = "##paragraph_end##`";
 
-            // Add space after the punctuation mark, except there is space/new line/quote after the punctuation mark
-            sanitizedString = Regex.Replace(sanitizedString, @"[\?\.\!\;\,\:](?![\n|""|'|\r|\r\n|\s+])", "$0 ");
-
             // if the input string is not markup, sanitize it
-            if (!tagRegex.IsMatch(sanitizedString))
+            if (!tagRegex.IsMatch(value))
             {
+                // Add space after the punctuation mark, except there is space/new line/quote after the punctuation mark
+                var sanitizedString = PunctuationRegex().Replace(value, "$0 ");
```

Comment:
We now will only sanitize non-html text.  Which could be problematic.  But currently it will corrupt valid URL values if we use it the way were were.